awx/main/models/unified_jobs.py

```diff
@@ -1512,3 +1512,77 @@ def log_lifecycle(self, state, blocked_by=None):
         else:
             msg = f"{self._meta.model_name}-{self.id} {state.replace('_', ' ')}"
         logger_job_lifecycle.debug(msg, extra=extra)
+
+    @property
+    def launched_by(self):
+        ancestor_job = self.ancestor_job
+        launched_by = {}
+        if ancestor_job.launch_type == "manual" or ancestor_job.launch_type == "relaunch":
+            user_obj=ancestor_job.created_by
+            launched_by['id'] = user_obj.id
+            launched_by['name'] = user_obj.username
+            launched_by['type'] = "User"
+            launched_by['url'] = "/api/v2/users/" + str(user_obj.id)
```

Comment:
Most objects in the system has a method you can call to generate this: `get_absolute_url`

```python
In [1]: User.objects.first().get_absolute_url()
Out[1]: '/api/v2/users/1/'
```

Reply:
100% this.